### PR TITLE
[Doc] [Connector-V2] Add connector-v2 kafka source doc

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request
issue:https://github.com/apache/incubator-seatunnel/issues/2840
Add connector-v2 kafka source and sink doc
sink doc in pr:https://github.com/apache/incubator-seatunnel/pull/2889

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: docs/en/connector-v2/source/kafka.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+- [x] [parallelism](../../concept/connector-v2-features.md)
+- [ ] [support user-defined split](../../concept/connector-v2-features.md)
+
+## Options
+
+| name              | type   | required | default value            |
+| ----------------- | ------ | -------- | ------------------------ |
+| topic             | String | yes      | -                        |
+| bootstrap.servers | String | yes      | -                        |
+| pattern           | String | no       | false                    |
```
Comment (hailin0) [inline docs/en/connector-v2/source/kafka.md]: String -> boolean?

File: docs/en/connector-v2/source/kafka.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+| consumer.group    | String | no       | SeaTunnel-Consumer-Group |
+| kafka.*           | String | no       | -                        |
+| common-options    |        | no       | -                        |
+
```
Comment (hailin0) [inline docs/en/connector-v2/source/kafka.md:29]: Add `commit_on_checkpoint` option?

https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java#L87

File: docs/en/connector-v2/source/kafka.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+| name              | type    | required | default value            |
+| ----------------- | ------- | -------- | ------------------------ |
+| topic             | String  | yes      | -                        |
+| bootstrap.servers | String  | yes      | -                        |
+| pattern           | Boolean | no       | false                    |
+| consumer.group    | String  | no       | SeaTunnel-Consumer-Group |
+| commit_on_checkpoint    | Boolean | no       | true |
+| kafka.*           | String  | no       | -                        |
+| common-options    |         | no       | -                        |
```
Comment (hailin0) [inline docs/en/connector-v2/source/kafka.md]: check style

**Review by hailin0 — COMMENTED** [review]
LGTM

**Review by hailin0 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]
+1

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**